### DISCUSSION TITLE
strongSwan destroys Phase 2 SAs when peer only sends delete for IKEv1 Phase 1 SA

### DISCUSSION THREAD

**Discussion by vmladd** [strongswan/strongswan#1085] [Q&A] [answered]
I have an IKEv1 tunnel between strongSwan and a device using the QuickSec IKE/IPsec library in which strongSwan is the responder.  On the initiator's side we configured a Phase 1 lifetime of 10 minutes and a Phase 2 lifetime of 3 minutes to force the initiator (and not strongSwan) to start any rekeys/reauthentications.  All of the Phase 2 rekeys occur successfully.

Our device uses a break-before-make strategy when performing the Phase 1 reauthentication.  At a high-level, we are observing the below steps (with the relevant timestamps from the strongSwan logs):
1. Our device sends a DEL only for the Phase 1 SA
2. Wed, 2022-06-08, 14:56:45: strongSwan receives the DEL and deletes the Phase 1 SA and all Phase 2 SAs
3. Wed, 2022-06-08, 14:56:55: Our device initiates a new Phase 1 SA which is successfully negotiated with strongSwan
4. Our device associates the existing Phase 2 SAs (which strongSwan deleted) with the newly negotiated Phase 1 SA.  We are now in a situation in which
  a. Our device has the new Phase 1 SA and the existing Phase 2 SAs
  b. strongSwan only has the Phase 1 SA
  c.  Traffic is dropped since strongSwan doesn't know about the Phase 2 SAs
6. Wed, 2022-06-08, 14:57:25: Our device starts a Phase 2 rekey which leads to a new set of Phase 2 SAs
7. Both sides are now consistent and traffic flows correctly

Per https://datatracker.ietf.org/doc/html/draft-jenkins-ipsec-rekeying-06#section-3.2, it seems like strongSwan should not alter the existing Phase 2 SAs when it receives the delete for the Phase 1 SA in step 2.
> Phase 1 SA Expiration:
      -DELETE notification should be sent for phase 1 SA only
      -phase 2 SAs between peers are left untouched

From the strongSwan changelog (https://wiki.strongswan.org/projects/strongswan/wiki/Changelog55), we also expected strongSwan to send delete notifications for the Phase 2 SAs when it removes them, but we do not see this occurring.
> A DELETE is sent when a rekeyed IKEv1 SA is deleted. This fixes issues with peers that continue
to send DPDs on the old SA and then delete all SAs if no response is received (see [#2090](https://wiki.strongswan.org/issues/2090)).
Also, when terminating IKEv1 SAs, DELETEs for all CHILD_SAs are now sent before sending one for
the IKE_SA and destroying it.

Can I get some help on the above two points on strongSwan's behavior?  This may be related to this thread (https://github.com/strongswan/strongswan/discussions/253) that did not have a resolution.  Please see below for details about the setup.

### ipsec.conf
```
# ipsec.conf - strongSwan IPsec configuration

config setup
      charondebug="all"
      uniqueids=never
      strictcrlpolicy=no

conn generic-tunnel
    auto=add
    esp=null-aes128-aes256-sha1-sha256-sha384-sha512-modp1024-modp1536-modp2048-modp3072-modp4096-modp6144,null-aes128-aes256-sha1-sha256-sha384-sha512
    ike=null-aes128-aes256-sha1-sha256-sha384-sha512-modp1024-modp1536-modp2048-modp3072-modp4096-modp6144
    fragmentation=yes
    type=tunnel

conn psk-tunnel
    also=generic-tunnel
    authby=psk
    leftauth=psk
    rightauth=psk
```

### ipsec statusall
```
Status of IKE charon daemon (strongSwan 5.6.2, Linux 5.2.2-050202-generic, x86_64):
  uptime: 57 minutes, since Jun 10 12:19:29 2022
  malloc: sbrk 2568192, mmap 0, used 738384, free 1829808
  worker threads: 11 of 16 idle, 5/0/0/0 working, job queue: 0/0/0/0, scheduled: 6
  loaded plugins: charon aesni aes rc2 sha2 sha1 md4 md5 mgf1 random nonce x509 revocation constraints pubkey pkcs1 pkcs7 pkcs8 pkcs12 pgp dnskey sshkey pem openssl fips-prf gmp agent xcbc hmac gcm attr kernel-netlink resolve socket-default connmark stroke updown eap-mschapv2 xauth-generic counters
Listening IP addresses:
  192.168.0.150
  169.254.6.18
  2222::1
  172.17.0.1
```

### charon_debug.log
[charon_debug.log](https://github.com/strongswan/strongswan/files/8881664/charon_debug.log)

### Our device
IP: 169.254.6.45
Phase 1 lifetime: 10 minutes
Phase 2 lifetime: 3 minutes

**Comment by tobiasbrunner** [comment, marked as answer]
Nothing will change about this. Stop using IKEv1!
